src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -227,6 +297,16 @@ TieredHNSWIndex<DataType, DistType>::~TieredHNSWIndex() {
             delete job;
         }
     }
+    // Delete all the pending repair jobs.
+    for (auto &jobs : this->idToRepairJobs) {
+        for (auto *job : jobs.second) {
+            delete job;
+        }
+    }
+    // Delete all the pending swap jobs.
+    for (auto *job : this->swapJobs) {
+        delete job;
+    }
```

Comment:
is this thread safe? Is that our responsibility or should a job delete itself if it realizes that the relevant index is deleted? This is what we implemented in the GC in RediSearch for reference

Reply:
It should be RediSearch responsibility to decrease the index ref count upon receiving `ft.drop` or something BEFORE this destructor is called. Then, when the jobs are executed asynchronously, they should first test if the index is alive, and only if so upgrade and execute the managed job callback that accesses this internal job pointer. EIther way, it should be our responsibility to release the pending jobs.

Reply:
No. I'll add one